src/checkbox/styles/checkbox.css

```diff
@@ -0,0 +1,78 @@
+@import '../../common/styles/variables.css';
+
+.label {
+	display: block;
+	font: var(--base-font);
+	margin-bottom: var(--line-spacing);
+	padding: 0.25em 0 0.25em 2em;
+	position: relative;
+}
+
+.root {
+	opacity: 0;
+	position: absolute;
+	z-index: -1;
+}
+
+.inputWrapper {
+	left: 0;
+	position: absolute;
+	top: 0;
+}
+
+.inputWrapper::before,
+.inputWrapper::after {
+	box-sizing: border-box;
+	content: '';
+	display: block;
+}
+
+.inputWrapper::before {
+	border: 2px solid var(--form-highlight-color);
+	height: 1.5em;
+	width: 1.5em;
+}
+
+.inputWrapper::after {
+	background-color: var(--form-highlight-color);
+	height: calc(1.5em - 6px);
+	left: 3px;
+	position: absolute;
+	top: 3px;
+	transform: scale(0);
+	transition: all var(--short-transition-duration) var(--short-transition-easing);
+	width: calc(1.5em - 6px);
+}
+
+/* checked style */
+.checked .inputWrapper::after {
+	transform: scale(1);
+}
+
+/* disabled and readonly styles */
+.disabled .inputWrapper::before,
+.readonly .inputWrapper::before {
+	border-color: var(--form-border-color);
+}
+
+/* required style, only applies to inputs with a label */
+.required span::after {
+	content: '*';
+	color: var(--dojo-error-color);
+}
+
+/* invalid style */
+.invalid .inputWrapper::before {
+	border-color: var(--dojo-error-color);
+}
+.invalid .inputWrapper::after {
+	background-color: var(--dojo-error-color);
+}
+
+/* valid style */
+.valid .inputWrapper::before {
```

Comment:
I don't care much either way, but it's worth mentioning that in our unidirectional flow paradigm, a parent component would have to explicitly set `valid` just like it would have to set `invalid`. Do we really think users will do this?

Either way, all form components should either support this or not to stay consistent.